source/blueprints/screening_app.rst

```diff
@@ -0,0 +1,507 @@
+
+   ic = nlp.domains['prediabetes_screening'].intent_classifier
+   ic.config.model_settings['classifier_type']
+
+.. code-block:: console
+
+   'logreg'
+
+.. code-block:: python
+
+   ic.config.features
+
+
+.. code-block:: console
+
+   {
+      'bag-of-words': {'lengths': [1, 2]}, 
+      'enable-stemming': True
+   }
+
+You can experiment with different learning algorithms (model types), features, hyperparameters, and cross-validation settings by passing the appropriate parameters to the classifier's :meth:`fit()` method. Here are a few examples.
+
+Experiment with the intent classifiers
+""""""""""""""""""""""""""""""""""""""
+
+We can change the feature extraction settings to use bag of trigrams in addition to the default bag of words:
+
+.. code:: python
+
+   ic.config.features['bag-of-words']['lengths'].append(3)
+   ic.fit()
+
+.. code-block:: console
+
+    Fitting intent classifier: domain='prediabetes_screening'
+    Selecting hyperparameters using k-fold cross-validation with 10 splits
+    Best accuracy: 98.40%, params: {'C': 1, 'class_weight': {0: 1.0488603988603988, 1: 1.0152380952380953, 2: 0.5086111111111111, 3: 1.1506472491909385, 4: 1.6275252525252526, 5: 1.1718076285240464, 6: 2.0179738562091503, 7: 0.7019113149847095, 8: 0.9538557213930349, 9: 3.804666666666667, 10: 1.6907407407407407, 11: 1.3684959349593495}, 'fit_intercept': True}
+
+Change the classification model to random forest instead of the default logistic regression:
+
+.. code:: python
+
+   ic.fit(model_settings={'classifier_type': 'rforest'}, param_selection={'type': 'k-fold', 'k': 10, 'grid': {'class_bias': [0.7, 0.3, 0]}})
+
+.. code-block:: console
+
+    Fitting intent classifier: domain='prediabetes_screening'
+    Selecting hyperparameters using k-fold cross-validation with 10 splits
+    Best accuracy: 96.90%, params: {'class_weight': {0: 1.0209401709401709, 1: 1.006530612244898, 2: 0.7894047619047618, 3: 1.0645631067961165, 4: 1.268939393939394, 5: 1.07363184079602, 6: 1.4362745098039216, 7: 0.8722477064220183, 8: 0.9802238805970149, 9: 2.202, 10: 1.2960317460317459, 11: 1.1579268292682925}}
+
+You can use similar options to inspect and experiment with the Entity Recognizer and the other NLP classifiers. Finding the optimal machine learning settings is a highly iterative process involving several rounds of model training (with varying configurations), testing, and error analysis. See the :doc:`User Guide <../userguide/nlp>` for more about training, tuning, and evaluating the various MindMeld classifiers.
+
+.. admonition:: Exercise
+
+   Experiment with different models, features, and hyperparameter selection settings to see how they affect classifier performance. Maintain a held-out validation set to evaluate your trained NLP models and analyze misclassified test instances. Then, use observations from the error analysis to inform your machine learning experimentation. See the :doc:`User Guide <../userguide/nlp>` for examples and discussion.
```

Comment:
"Maintain a held-out validation set to evaluate your trained NLP models and analyze misclassified ~~test~~ instances."

Just so we are clear that we only analyze errors on the validation set and not the test set.